src/Solver/Dynamic/Ida.cpp

```diff
@@ -517,12 +506,17 @@ namespace AnalysisManager
     template <class ScalarT, typename IdxT>
     int Ida<ScalarT, IdxT>::deleteAdjoint()
     {
+      N_VDestroy(yyB_);
+      N_VDestroy(ypB_);
+      N_VDestroy(qB_);
+      SUNLinSolFree(linearSolverB_);
+      SUNMatDestroy(JacobianMatB_);
```

Comment:
I guess I didn't document this well. Function `deleteAdjoint()` is only undoing adjoint initialization, so that adjoint can be initialized for a new number of backward integration points. Vectors `yyB_`, `ypB_`, etc., will be reused. Deleting them here causes a segfault at the second step of a dynamic constrained optimization. See for example this [function](https://github.com/ORNL/GridKit/blob/develop/src/Solver/Optimization/DynamicObjective.cpp#L103).
```suggestion
```
I am not sure how is this passing CI tests.

Reply:
> I am not sure how is this passing CI tests.

CI does not build Ipopt because of the CoinHSL dependency. So, those 4 failing tests are not being tested in CI.